src/h5web/providers/my-utils.ts

```diff
@@ -2,25 +2,32 @@ import { nanoid } from 'nanoid';
 import {
   HDF5Attribute,
   HDF5Dims,
+  HDF5ExternalLink,
+  HDF5LinkClass,
   HDF5NumericType,
+  HDF5ScalarShape,
   HDF5Shape,
   HDF5SimpleShape,
+  HDF5StringType,
   HDF5Type,
   MyHDF5Dataset,
   MyHDF5Datatype,
   MyHDF5Entity,
   MyHDF5EntityKind,
   MyHDF5Group,
+  MyHDF5Link,
 } from '../providers/models';
 import { makeSimpleShape, makeStrAttr } from '../providers/mock/data';
-import { NxInterpretation } from '../visualizations/nexus/models';
+import { NxInterpretation, SilxStyle } from '../visualizations/nexus/models';
 
-type Opts = Partial<Pick<MyHDF5Dataset, 'id' | 'attributes'>>;
+type EntityOpts = Partial<Pick<MyHDF5Dataset, 'id' | 'attributes'>>;
+type GroupOpts = EntityOpts & { children?: MyHDF5Entity[] };
+type AllOrNone<T> = T | { [K in keyof T]?: never };
```

Comment:
I use this to type `makeMyNxDataGroup` to enforce that both or neither of `axes` and `axesAttr` be provided.

Reply:
Could it be replaced by supplying two overrides of `makeMyNxDataGroup` (with and, without the `axes` and `axesAttr` ?)

Reply:
Yeah, but the signature is super long, so not sure it's more readable...

Reply:
I've tried a few things, but nothing else worked and/or looked clearer.

Reply:
I should clarify that it works with two overrides, but then ESLint complains that they can be combined, but when I do, the typing of `axesAttr` stops working 😢

Reply:
:shrug: